snakemake_storage_plugin_irods/__init__.py

```
@@ -231,13 +237,29 @@ def __post_init__(self):
+        if self.settings.password is None:
+            irodsA = os.path.expanduser("~/.irods/.irodsA")
+            try:
+                with open(irodsA, "r") as r:
+                    scrambled_password = r.read()
+                    password = password_obfuscation.decode(scrambled_password)
+                    authentication_scheme = "native"
+            except OSError:
+                raise Exception(
+                    "Error: could not retrieve irods_password from"
+                    + " settings or ~/.irods/.irodsA file."
+                )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)